pkg/multicluster/register/registrant.go

```diff
@@ -620,38 +622,50 @@ func (c *clusterRegistrant) getTokenForSa(
 }
 
 // if:
-//   * we are operating against a context named "kind-", AND
 //   * the server appears to point to localhost, AND
 //   * the --local-cluster-domain-override flag is populated with a value
 //
-// then we rewrite the server config to communicate over the value of `--local-cluster-domain-override`, which
-// resolves to the host machine of docker. We also need to skip TLS verification
-// and zero-out the cert data, because the cert on the remote cluster's API server wasn't
-// issued for the domain contained in the value of `--local-cluster-domain-override`.
-//
+// then we need to skip TLS verification  and zero-out the cert data, because the cert
+// on the remote cluster's API server wasn't issued for the domain contained in the
+// value of `--local-cluster-domain-override`.
 // this function call is a no-op if those conditions are not met
-func (c *clusterRegistrant) hackClusterConfigForLocalTestingInKIND(
+func (c *clusterRegistrant) skipTLSVerificationForLocalTesting(
 	remoteCluster *api.Cluster,
-	remoteContextName string,
 ) error {
 	serverUrl, err := url.Parse(remoteCluster.Server)
 	if err != nil {
 		return err
 	}
 
-	if strings.HasPrefix(remoteContextName, "kind-") &&
-		(serverUrl.Hostname() == "127.0.0.1" || serverUrl.Hostname() == "localhost") &&
-		c.localAPIServerAddress != "" {
+	if c.localAPIServerAddress != "" && (serverUrl.Hostname() == "127.0.0.1" || serverUrl.Hostname() == "localhost") {
+		remoteCluster.InsecureSkipTLSVerify = true
+		remoteCluster.CertificateAuthority = ""
+		remoteCluster.CertificateAuthorityData = []byte("")
+	}
 
-		port := serverUrl.Port()
+	return nil
+}
+
+// if:
+//   * the --local-cluster-domain-override flag is populated with a value
+//
+// then rewrite the server config to communicate over the value of
+// `--local-cluster-domain-override`, which resolves to the host machine of docker.
+// There are use cases where the address a user uses to communicate with kubernetes from
+// their local machine may differ from the one used by pods running in the cluster.
+func (c *clusterRegistrant) respectApiServerOverride(remoteCluster *api.Cluster) error {
```

Comment:
nit: "respect" doesn't have a very clear meaning in this context, what about simply `set` instead of `respect`?